Yoast/wordpress-seo
# Loads the linkwatcher also on the frond end.

## Summary

This PR can be summarized in the following changelog entry:

* Fixes a bug where the text link counter doesn't count links for scheduled posts when they get published.
* Fixes a bug where the internal link count is not updated when a post is removed permanently.

## Relevant technical choices:

* Moved to a place where the hooks also get loaded on the frontend.
* Used the `transition_post_status` hook for catching a post status transition.
* Removed the links when a post isn't published anymore.

## Test instructions

This PR can be tested by following these steps:

* Checkout trunk. Make a scheduled post (like 2 minutes later) that contains a link to another post.
* Go to the homepage and wait for the post popping up.
* Go to the post overview in the admin. See the links aren't counted. A reindex is needed.
* Checkout this branch and follow the same steps.

Deleting a post:
* Deleting a post updates the number of referenced links for the other posts that linked to the post properly with this PR.

Fixes #7486
